docs/guide/rl_tips.rst

```diff
@@ -0,0 +1,241 @@
+
+To sum it up:
+
+Discrete Actions
+----------------
+
+.. note::
+
+	This covers `Discrete`, `MultiDiscrete`, `Binary` and `MultiBinary` spaces
+
+
+Discrete Actions - Single Process
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+DQN with extensions (double DQN, prioritized replay, ...) and ACER are the recommended algorithms.
+DQN is usually slower to train (regarding wall clock time) but is the most sample efficient (because of its replay buffer).
+
+Discrete Actions - Multiprocessed
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You should give a try to PPO2, A2C and its successors (ACKTR, ACER).
+
+If you can multiprocess the training using MPI, then you should checkout PPO1 and TRPO.
+
+
+Continuous Actions
+------------------
+
+Continuous Actions - Single Process
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Current State Of The Art (SOTA) algorithms are `SAC` and `TD3`.
+Please use the hyperparameters in the `RL zoo <https://github.com/araffin/rl-baselines-zoo>`_ for best results.
+
+
+Continuous Actions - Multiprocessed
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Take a look at PPO2, TRPO or A2C. Again, don't forget to take the hyperparameters from the `RL zoo <https://github.com/araffin/rl-baselines-zoo>`_
+for continuous actions problems (cf *Bullet* envs).
+
+.. note::
+
+  Normalization is critical for those algorithms
+
+If you can use MPI, then you can choose between PPO1, TRPO and DDPG.
+
+
+Goal Environment
+-----------------
+
+If your environment follows the `GoalEnv` interface (cf `HER <her.html>`_), then you should use
+HER + (SAC/TD3/DDPG/DQN) depending on the action space.
+
+
+.. note::
+
+	The number of workers is an important hyperparameters for experiments with HER. Currently, only HER+DDPG supports multiprocessing using MPI.
+
+
+
+Tips and Tricks when creating a custom environment
+==================================================
+
+If you want to learn about how to create a custom environment, we recommend you to read this `page <custom_envs.html>`_.
+We also provide a `colab notebook <https://colab.research.google.com/github/araffin/rl-tutorial-jnrr19/blob/master/5_custom_gym_env.ipynb>`_ for
+a concrete example of creating a custom gym environment.
+
+Some basic advice:
+
+- always normalize your observation space when you can, i.e., when you know the boundaries
+- normalize your action space and make it symmetric when continuous (cf potential issue below) A good practice is to rescale your actions to lie in [-1, 1]. This does not limit you as you can easily rescale the action inside the environment
+- start with shaped reward (i.e. informative reward) and simplified version of your problem
+- debug with random actions to check that your environment works and follows the gym interface:
+
+
+Here is a code snippet to check that your environment runs without error.
+
+.. code-block:: python
+
+	env = YourEnv()
+	obs = env.reset()
+	n_steps = 10
+	for _ in range(n_steps):
+	    # Random action
+	    env = env.action_space.sample()
```

Comment:
This won't run. It should be `env = env.action_space.sample()`

Reply:
sorry, I meant to say `action = env.action_space.sample()`

Reply:
nice catch, thanks